test/e2e/operator.go

```diff
@@ -272,3 +276,108 @@ var _ = Describe("ARO Operator - Conditions", func() {
 		Expect(err).NotTo(HaveOccurred())
 	})
 })
+
+var _ = Describe("ARO Operator - Azure NSG Reconciler", func() {
+	var vnetName, location, resourceGroup, workerSubnet, masterSubnet string
+	var dummynsg mgmtnetwork.SecurityGroup
+	ctx := context.Background()
+	enableReconcileNSG := func() {
+		instance, err := clients.AROClusters.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		if !instance.Spec.Features.ReconcileNSGs {
+			instance.Spec.Features.ReconcileNSGs = true
+			_, err = clients.AROClusters.AroV1alpha1().Clusters().Update(ctx, instance, metav1.UpdateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		}
+	}
+	gatherNetworkInfo := func() {
+		oc, err := clients.OpenshiftClustersv20200430.Get(ctx, vnetResourceGroup, clusterName)
+		Expect(err).NotTo(HaveOccurred())
+		vnetID, _, err := subnet.Split(*oc.OpenShiftClusterProperties.MasterProfile.SubnetID)
+		Expect(err).NotTo(HaveOccurred())
+
+		r, err := azure.ParseResourceID(vnetID)
+		Expect(err).NotTo(HaveOccurred())
+		resourceGroup = r.ResourceGroup
+
+		vnet, err := clients.VirtualNetworks.Get(ctx, resourceGroup, r.ResourceName, "")
+		Expect(err).NotTo(HaveOccurred())
+		location = *vnet.Location
+		vnetName = *vnet.Name
+
+		_, masterSubnet, err = subnet.Split(strings.ToLower(*oc.OpenShiftClusterProperties.MasterProfile.SubnetID))
+		Expect(err).NotTo(HaveOccurred())
+		_, workerSubnet, err = subnet.Split(strings.ToLower(*(*oc.OpenShiftClusterProperties.WorkerProfiles)[0].SubnetID))
```

Comment:
I think there is no need to do `strings.ToLower` here. I guess it was a copy-paste from `[Admin API] List Azure resources action` where it is important for deduplication reasons. In our case it is not required, I believe.